src/export/StructureDefinitionExporter.ts

```diff
@@ -364,10 +357,32 @@ export class StructureDefinitionExporter implements Fishable {
       delete elements[0].extension;
     }
 
+    // Hack Alert:
+    // A material change is required to an element AFTER the execution of the
+    // structDef.captureOriginalElements() function to ensure the differential
+    // is properly rendered.
+    // To ensure the differential root element contains only the appropriate
+    // attributes, we are setting an initial value for the prohibited "maxLength"
+    // element attribute that will then be deleted below. The "maxLength"
+    // attribute was selected because it is prohibited in type definitions (i.e.,
+    // logical models and resources).
+    // Ref: http://hl7.org/fhir/elementdefinition.html#interpretation
```

Comment:
Clever approach.  I think this works...

I know you considered some other approaches.  Two that I can think of are:
1. `StructureDefinition.toJSON`: As it loops through the elements checking `hasDiff()` (to generate the differential), could it also check if it is the element at index 0 in a specialization?
2. `Element.hasDiff`: Could it check if it is the root element in a specialization and return true if so?

I think I like option 1 above best.  Are there reasons this would not be sufficient?

Reply:
Thank you! I hate hacks! I guess I had some tunnel vision. I agree approach #1 is the better option, so I've implemented that approach and removed the hack.